File: Partake/app/in/partake/controller/api/debug/ErrorDBAPI.java
package in.partake.controller.api.debug;

import play.mvc.Result;
import in.partake.base.PartakeException;
import in.partake.controller.api.AbstractPartakeAPI;
import in.partake.model.dao.DAOException;
import in.partake.resource.ServerErrorCode;

public class ErrorDBAPI extends AbstractPartakeAPI {

    public static Result get() throws DAOException, PartakeException {
        return new ErrorDBAPI().execute();
    }

    /**
     * データベースエラー。
     * ステータスは 500 を返す。
     */
    @Override
    protected Result doExecute() throws DAOException {
        return renderError(ServerErrorCode.DB_ERROR);
    }

}

File: app/controllers/api/debug/ErrorDBAPI.scala
package controllers.api.debug

import controllers.api.AbstractAPI
import controllers.ActionContext
import in.partake.resource.ServerErrorCode
import play.api.mvc.Request
import play.api.mvc.AnyContent
import play.api.mvc.PlainResult

/**
 * データベースエラー。
 * ステータスは 500 を返す。
 */
object ErrorDBAPI extends AbstractAPI[Unit, Unit] {
  override def parseRequest(request: Request[AnyContent])(implicit context: ActionContext): Unit = {
  }

  override def executeAction(params: Unit)(implicit context: ActionContext): Unit = {
  }

  override def renderResult(values: Unit)(implicit context: ActionContext): PlainResult = {
    return renderError(ServerErrorCode.DB_ERROR);
  }
}
